### PR TITLE
[LTR] Implement Gandalf, White Rider

### PR DESCRIPTION
**Pull request by TiagoMDG**
* Took ideas from the God Eternals, Arwen Undomiel and Enigma Sphinx.
* Tested it with multiple interactions, including murdering a stolen Gandalf with Agent of Treachery.

### CODE REVIEW

File: Mage.Sets/src/mage/cards/g/GandalfWhiteRider.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                new BoostControlledEffect(1, 0, Duration.EndOfTurn).setText("creatures you control get +1/+0 until end of turn"),
+                StaticFilters.FILTER_SPELL_A,
+                false
+        );
+
+        // Add scry effect to ability
+        ability.addEffect(new ScryEffect(1, false));
+
+        this.addAbility(ability);
+        this.addAbility(new GandalfWhiteRiderTriggeredAbility(new GandalfWhiteRiderDyingEffect(), true));
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/g/GandalfWhiteRider.java:56]: Why not use `DiesSourceTriggeredAbility` ?

File: Mage.Sets/src/mage/cards/g/GandalfWhiteRider.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        return new GandalfWhiteRiderDyingEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+        Card card = (Card) source.getSourceObjectIfItStillExists(game);
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/g/GandalfWhiteRider.java:129]: This doesn't look right... need to check the zcc (compare to the "permanentLeftBattlefield" value stored from DiesSourceTriggeredAbility)

File: Mage.Sets/src/mage/cards/g/GandalfWhiteRider.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+
+        // Vigilance
+        this.addAbility(VigilanceAbility.getInstance());
+
+        // Whenever you cast a spell, creatures you control get +1/+0 until end of turn. Scry 1.
+        // When Gandalf, White Rider dies, you may put it into its owner's library fifth from the top.
+
+        // Adds ability that whenever you cast a spell it gives a temporary +1/+0 effect until end of turn
+        Ability ability = new SpellCastControllerTriggeredAbility(
+                new BoostControlledEffect(1, 0, Duration.EndOfTurn).setText("creatures you control get +1/+0 until end of turn"),
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/g/GandalfWhiteRider.java:47]: nit, is there a reason this text needs to be set manually? seems straightforward enough that auto generation should be correct

**Review by xenohedron — CHANGES_REQUESTED** [review]

**Comment by TiagoMDG** [comment]
Going to push the zcc comparison in a bit.

**Review by xenohedron — APPROVED** [review]

**Comment by xenohedron** [comment]
> Going to push the zcc comparison in a bit.

oh right good call

**Pull request merged**